<a href="https://colab.research.google.com/github/VennelaPulavarti/NNandDLclass5/blob/main/NN%26DL_ICP_5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [12]:
#Importing packages

import pandas as pd
import numpy as np

from sklearn.naive_bayes import GaussianNB
from sklearn.svm import SVC, LinearSVC

from sklearn import metrics
from sklearn.metrics import accuracy_score
from sklearn.model_selection import train_test_split

In [2]:
#loading data set
df = pd.read_csv("glass.csv")

df.head()

RI     Na    Mg    Al     Si     K    Ca   Ba   Fe  Type
0  1.52101  13.64  4.49  1.10  71.78  0.06  8.75  0.0  0.0     1
1  1.51761  13.89  3.60  1.36  72.73  0.48  7.83  0.0  0.0     1
2  1.51618  13.53  3.55  1.54  72.99  0.39  7.78  0.0  0.0     1
3  1.51766  13.21  3.69  1.29  72.61  0.57  8.22  0.0  0.0     1
4  1.51742  13.27  3.62  1.24  73.08  0.55  8.07  0.0  0.0     1

In [3]:
#removing the column to be predicted from the dataframe
X2 = df.drop(["Type"],axis=1)

#creating an array for the target column prediction
y = df["Type"]

In [4]:
#splitting data(both X,y) for train & for test

X_train, X_test, y_train, y_test = train_test_split(X2, y, test_size= 0.8 , random_state = 0)

In [14]:
# QUESTION 1 - NAIVE BAYES METHOD

#Training data with Gaussian Naive Bayes
gnb = GaussianNB()

# Training on X_Train & y_Train and Applying prediction on X_Test
y_pred = gnb.fit(X_train, y_train).predict(X_test)

# calculating the accuracy score (accuracy score is calculated on predicted value & Actual value)
acc_nb = accuracy_score(y_test,y_pred)
print("Naive Bayes Accuracy :", acc_nb)

#Calculating %
print("Naive Bayes Accuracy in percentage :")
round(acc_nb * 100,2)

Naive Bayes Accuracy : 0.5348837209302325
Naive Bayes Accuracy in percentage :


53.49

In [15]:
# Classification report
print(metrics.classification_report(y_test,y_pred, zero_division=0))

              precision    recall  f1-score   support

           1       0.58      0.23      0.33        60
           2       0.45      0.72      0.55        61
           3       0.00      0.00      0.00        15
           5       0.47      0.88      0.61         8
           6       0.55      1.00      0.71         6
           7       0.88      0.95      0.91        22

    accuracy                           0.53       172
   macro avg       0.49      0.63      0.52       172
weighted avg       0.52      0.53      0.48       172



In [16]:
# QUESTION 2 - LINEAR SVM METHOD

#Training data with Support vector Classification
svc = SVC(kernel ='linear', C = 5)

# Training on X_Train & y_Train
svc.fit(X_train, y_train)

#Applying prediction on X_Test
y_pred = svc.predict(X_test)

# calculating the accuracy score(accuracy score is calculated on predicted value & Actual value)
acc_svc=accuracy_score(y_test,y_pred)
print("LinearSVM Accuracy is:", acc_svc)

# Calculating %
print("SVM Accuracy in percentage :")
round(acc_svc * 100 , 2)

LinearSVM Accuracy is: 0.622093023255814
SVM Accuracy in percentage :


62.21

In [10]:
# Classification report is used to measure the quality of predictions using classification algorithm and accuracy
print(metrics.classification_report(y_test,y_pred, zero_division=0))

              precision    recall  f1-score   support

           1       0.67      0.62      0.64        60
           2       0.54      0.67      0.60        61
           3       0.00      0.00      0.00        15
           5       0.47      0.88      0.61         8
           6       0.67      0.67      0.67         6
           7       0.95      0.82      0.88        22

    accuracy                           0.62       172
   macro avg       0.55      0.61      0.57       172
weighted avg       0.59      0.62      0.60       172



 # In Gaussian Naive Bayes accuracy score is 53.49%
 # In Support vector Classification accuracy score is 62.61%

 # In this data set, Linear SVM is better than Gaussian Naive Bayes,
# as accuracy score of Linear Svm is highest.